### PR TITLE
Feature/55 reconsider UUID

### PR DESCRIPTION
**Pull request by timohuber**

### CODE REVIEW

File: pool/app/session/event.ml [hunk 1]
```diff
@@ -30,8 +30,7 @@ type reschedule =
 
 (* TODO [aerben] experiment ID *)
```
Comment (mabiede) [inline pool/app/session/event.ml:31]: remove todo ;-)

File: pool/app/session/repo/repo_entity.ml [hunk 2]
```diff
@@ -1,11 +1,21 @@
-module Id = Pool_common.Id
 module RepoId = Pool_common.Repo.Id
 module Reminder = Pool_common.Reminder
 module RepoReminder = Pool_common.Repo.Reminder
 
+module Id = struct
+  include Entity.Id
+
+  let t =
+    Pool_common.Repo.make_caqti_type
```
Comment (mabiede) [inline pool/app/session/repo/repo_entity.ml:9]: use `include Pool_common.Repo.Id` which includes this function already

File: pool/app/session/session.mli [hunk 3]
```diff
@@ -236,6 +255,12 @@ val has_bookable_spots_for_experiments
   -> Experiment.Id.t
   -> (bool, Pool_common.Message.error) result Lwt.t
 
+module Repo : sig
```
Comment (mabiede) [inline pool/app/session/session.mli:258]: I didn't find the usage outside of the module, where this is used.
Can this be removed?

File: pool/app/session/session.mli [hunk 4]
```diff
@@ -115,10 +135,9 @@ val has_assignments : t -> bool
 val session_date_to_human : t -> string
 
 (* TODO [aerben] this should be experiment id type *)
```
Comment (mabiede) [inline pool/app/session/session.mli:137]: todos can be removed

File: pool/app/session/session.ml [hunk 5]
```diff
@@ -24,3 +24,7 @@ let has_bookable_spots_for_experiments tenant experiment =
   >|+ CCList.is_empty
   >|+ not
 ;;
+
+module Repo = struct
```
Comment (mabiede) [inline pool/app/session/session.ml:28]: Does this need to be exposed?

File: pool/app/session/session.mli [hunk 6]
```diff
@@ -166,12 +185,12 @@ val assignment_creatable : t -> (unit, Pool_common.Message.error) result
...
```
Comment (mabiede) [inline pool/app/session/session.mli:185]: todo can be removed

File: pool/test/integration_utils.ml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            ; lastname = Contact.lastname contact
+            ; terms_accepted_at = None
+            ; language = contact.language
+            })
+      ]
+      @ verified
+      |> Lwt_list.iter_s (Contact.handle_event Data.database_label)
+    in
+    contact
+    |> Contact.id
```
Comment (mabiede) [inline pool/test/integration_utils.ml:39]: How about opening `Contact` in this function, it's used many times.

**Review by mabiede — CHANGES_REQUESTED** [review]

File: pool/app/experiment/repo/repo.ml [hunk 8]
```diff
@@ -142,7 +142,7 @@ module Sql = struct
     let open Caqti_request.Infix in
     {sql|
       INNER JOIN pool_mailing
-        ON pool_experiments.id = pool_mailing.experiment_id
+        ON pool_experiments.uuid = pool_mailing.experiment_uuid
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/app/experiment/repo/repo.ml:145]: Maybe 😄 lots of sources say that joins are faster most of the time. But always using examples retrieving multiple rows. 

I changed this one.

File: pool/app/invitation/repo/repo.ml [hunk 9]
```diff
@@ -180,8 +166,8 @@ module Sql = struct
       FROM
         pool_invitations
       WHERE
-        (SELECT id FROM pool_experiments WHERE pool_experiments.uuid = UNHEX(REPLACE($1, '-', ''))),
-        (SELECT id FROM pool_contacts WHERE pool_contacts.user_uuid = UNHEX(REPLACE($2, '-', '')))
+        pool_invitations.experiment_uuid = UNHEX(REPLACE($1, '-', '')),
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/app/invitation/repo/repo.ml:169]: No that is not. 

This function is unused, I removed it.

File: pool/test/integration_utils.ml (hunk 7 above)
Comment (timohuber) [reply pool/test/integration_utils.ml:39]: Cannot be opened before `let contact = Model.create_contact ?id ~with_terms_accepted () in` (Line 14) due to naming conflicts.
But yes

File: pool/app/experiment/repo/repo.ml (hunk 8 above)
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/experiment/repo/repo.ml]: Thanks. Yes, I was wondering if the selected value of the separate table is unique.

**Review by mabiede — APPROVED** [review]

**Pull request merged**